internal/pendingpool/pendingpool.go

```diff
@@ -0,0 +1,85 @@
+package pendingpool
+
+import (
+	"database/sql"
+	"encoding/hex"
+	"errors"
+
+	"github.com/SIGBlockchain/project_aurum/internal/constants"
+	"github.com/SIGBlockchain/project_aurum/internal/producer/src/contracts"
+	"github.com/SIGBlockchain/project_aurum/internal/producer/src/hashing"
+	"github.com/SIGBlockchain/project_aurum/internal/producer/src/validation"
+	"github.com/SIGBlockchain/project_aurum/internal/publickey"
+)
+
+//PendingData contains pending balance and pending nonce
+type PendingData struct {
+	PendingBal   uint64
+	PendingNonce uint64
+}
+
+//PendingMap contains a map that maps a hex encoded string of a wallet address to a pointer of PendingData
+type PendingMap struct {
+	Sender map[string]*PendingData
+}
+
+//NewPendingData returns an instance of pendingData given pending balance and pending nonce
+func NewPendingData(pendingBal uint64, pendingNonce uint64) PendingData {
+	return PendingData{pendingBal, pendingNonce}
+}
+
+//NewPendingMap returns an instance of pendingMap given a wallet address and an instance of pendingData
+func NewPendingMap(walletAddress []byte, data PendingData) PendingMap {
+	encodedWalletAddr := hex.EncodeToString(walletAddress)
+	m := make(map[string]*PendingData)
+	m[encodedWalletAddr] = &data
+	return PendingMap{m}
+}
+
+//Add returns an error if the process of validating the given contract has failed.
+//Otherwise, Add either inserts the sender's PKhash and the PendingData struct into the map,
+//or updates the pending balance and pending nonce for that sender's PKhash in the map
+func (m *PendingMap) Add(c *contracts.Contract) error {
+	senderPKHash := hashing.New(publickey.Encode(c.SenderPubKey))
+	senderPKStr := hex.EncodeToString(senderPKHash) // hex encoded sender PKhash string for the key
+	if m.Sender == nil {
```

Comment:
Isn't this redundant if the map is already constructed in the `NewPendingMap` function?